tests/controllers/test_uaa_controller.py

```diff
@@ -140,3 +167,96 @@ def test_get_groups_failure(self, mock_request):
         with self.app.test_request_context():
             with self.assertRaises(HTTPError):
                 uaa_controller.get_groups()
+
+    # create_user_account_with_random_password
+
+    @requests_mock.mock()
+    def test_create_user_account_with_random_password_success(self, mock_request):
+        mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
+        mock_request.post(url_uaa_users, json=create_user_success_json, status_code=200)
+        with self.app.test_request_context():
+            self.assertEqual(
+                uaa_controller.create_user_account_with_random_password(user_email, user_first_name, user_last_name),
+                create_user_success_json,
+            )
+
+    @requests_mock.mock()
+    def test_create_user_account_with_random_password_user_already_exists(self, mock_request):
+        mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
+        mock_request.post(url_uaa_users, json=create_user_already_exists_json, status_code=409)
+        with self.app.test_request_context():
+            self.assertEqual(
+                uaa_controller.create_user_account_with_random_password(user_email, user_first_name, user_last_name),
+                {"error": "Username already in use: some.one@ons.gov.uk"},
+            )
+
+    # change_user_password
+
+    @requests_mock.mock()
+    def test_change_user_password_by_id_user_not_found(self, mock_request):
+        mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
+        mock_request.get(url_uaa_user_by_email, status_code=404)
+
+        with self.app.test_request_context():
+            self.assertIsNone(uaa_controller.change_user_password("some.one@ons.gov.uk", user_password))
+
+    @requests_mock.mock()
+    def test_change_user_password_retrieve_code_failure(self, mock_request):
+        mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
+        mock_request.get(url_uaa_user_by_email, json=uaa_user_search_email, status_code=200)
+        # Docs aren't clear about failure codes for this endpoint so 500 is a safe one we can assume it can throw.
```

Comment:
We need to migrate away from 500, but I realise this is everywhere

Reply:
This is just the error code from uaa I'm talking about.  The API isn't clear what error status codes it can throw so I had to guess a 500.  But I do agree that anything front facing shouldn't be 500!